### PR TITLE
Added a new test for phptravels.com

### PR DESCRIPTION
**Pull request by Zoltekk**
A test which logs in, navigates to the hotels subpage and tests the inputs for rooms in the search bar

### CODE REVIEW

File: phptravels website/config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (Glovacky) [inline phptravels website/config.json]: Please remove all *.pyc files. They are not supposed to be pushed to the repo.

File: phptravels website/task_1.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+go to phptravels.net and register to get credentials.
```
Comment (Glovacky) [inline phptravels website/task_1.txt]: please remove website/geckodriver.log

File: phptravels website/test_hotel_subpage.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+# 1. Log in
+# 2. Navigate to "Hotels" subpage
+# 3. Test different boundary values in the search element
+
+
+def room_fillout(driver, get_xpaths, options):
+    room_choice = driver.find_element(By.XPATH, get_xpaths["subpages"]["room_choice"])
+    room_choice.click()
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, get_xpaths["subpages"]["room_inc"])))
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py:17]: explicit wait doesn't seem to be necessary

File: phptravels website/test_hotel_subpage.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+@pytest.mark.parametrize("childs", [-1, "abc", 2, 5000])
+def test_boundary(get_login, get_settings, get_xpaths, get_cred, rooms, adults, childs):
+
+    driver = get_login
+
+    time.sleep(2)
+    hotels_sub = driver.find_element(By.XPATH, get_xpaths["subpages"]["hotels_sub"])
+    hotels_sub.click()
+
+    city_box = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, get_xpaths["subpages"]["city_box"])))
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py]: is this necessary?

File: phptravels website/test_hotel_subpage.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    ActionChains(driver).move_to_element(city_box).click(city_box).send_keys("Lon").perform()
+
+    time.sleep(3)
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py]: is this necessary?

File: phptravels website/test_hotel_subpage.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    city_choice = driver.find_element(By.XPATH, get_xpaths["subpages"]["city_choice"])
+    ActionChains(driver).move_to_element(city_choice).send_keys(Keys.RETURN).perform()
+
+    options = {"rooms": rooms, "adults": adults, "childs": childs}
+    room_fillout(driver, get_xpaths, options)
+
+    submit_button = driver.find_element(By.XPATH, get_xpaths["subpages"]["submit_button"])
+    submit_button.click()
+    time.sleep(3)
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py]: is this necessary?

File: phptravels website/conftest.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+def get_login(get_driver, get_settings, get_xpaths, get_cred):
+    driver = get_driver
+    driver.get(get_settings["base_url"])
+    login = driver.find_element(By.XPATH, get_xpaths["login"]["username_input"])
+    login.send_keys(get_cred["login"])
+
+    password = driver.find_element(By.XPATH, get_xpaths["login"]["password_input"])
+    password.send_keys(get_cred["password"])
+
+    login_button = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, get_xpaths["login"]["login_button"])))
```
Comment (Glovacky) [inline phptravels website/conftest.py]: is this necessary?

File: phptravels website/test_hotel_subpage.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+
+@pytest.mark.parametrize("rooms", [-1, "abc", 0, 2, 20])
+@pytest.mark.parametrize("adults", [-1, "abc", 0, 2, 5000])
+@pytest.mark.parametrize("childs", [-1, "abc", 2, 5000])
+def test_boundary(get_login, get_settings, get_xpaths, get_cred, rooms, adults, childs):
+
+    driver = get_login
+
+    time.sleep(2)
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py]: let's try not to hard code time.sleep() unless it's completely necesssary

File: phptravels website/config.json [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+      "city_box": "//span[@id='select2-hotels_city-container']",
+      "city_choice": "//ul[@id='select2-hotels_city-results']/li",
+      "room_choice": "//a[@data-toggle='dropdown']",
+      "room_inc": "//div[@class='roomInc']",
+      "submit_button": "//button[@id='submit']"
+    }
+  },
+  "credentials": {
+    "login": "user@phptravels.com",
+    "password": "demouser"
```
Comment (Glovacky) [inline phptravels website/config.json]: better not to push credentials into repo. Try using environment variables and dotenv library.

File: phptravels website/test_hotel_subpage.py [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def room_fillout(driver, get_xpaths, options):
+    room_choice = driver.find_element(By.XPATH, get_xpaths["subpages"]["room_choice"])
+    room_choice.click()
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, get_xpaths["subpages"]["room_inc"])))
+
+    for key, value in options.items():
+        driver.find_element(By.XPATH, "//input[@id='" + key + "']").clear()
+        driver.find_element(By.XPATH, "//input[@id='" + key + "']").send_keys(options[key])
```
Comment (Glovacky) [inline phptravels website/test_hotel_subpage.py]: 1. Instead of finding the same element twice you can just find it once, store it in a variable and perform both clear() and send_keys()
2. Put that XPATH in a json file. You can use %s for concatenation

**Review by Glovacky — CHANGES_REQUESTED** [review]

File: phptravels website/test_hotel_subpage.py (hunk 3 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py:17]: There's an animation when clicking the room_choice element, the wait is there to hold out until the next element is visible

File: phptravels website/test_hotel_subpage.py (hunk 4 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py]: Same as above

File: phptravels website/test_hotel_subpage.py (hunk 5 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py]: The website takes a moment to load in the list of available cities, same as above

File: phptravels website/test_hotel_subpage.py (hunk 6 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py]: Removed

File: phptravels website/conftest.py (hunk 7 above)
Comment (Zoltekk) [reply phptravels website/conftest.py]: Removed

File: phptravels website/test_hotel_subpage.py (hunk 8 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py]: Removed

File: phptravels website/config.json (hunk 1 above)
Comment (Zoltekk) [reply phptravels website/config.json]: Removed

File: phptravels website/task_1.txt (hunk 2 above)
Comment (Zoltekk) [reply phptravels website/task_1.txt]: Removed

File: phptravels website/config.json (hunk 9 above)
Comment (Zoltekk) [reply phptravels website/config.json]: Changed

File: phptravels website/test_hotel_subpage.py (hunk 10 above)
Comment (Zoltekk) [reply phptravels website/test_hotel_subpage.py]: Changed

**Review by Glovacky — APPROVED** [review]
